client/src/components/History/model/syncCurrentHistoryToGalaxy.js

```diff
@@ -1,28 +1,23 @@
 // Sync Galaxy store to legacy galaxy current history
 
-import { getGalaxyInstance } from "app";
-import { switchMap, pluck } from "rxjs/operators";
-import { monitorChange } from "utils/observable/monitorChange";
+import { switchMap, pluck, shareReplay, filter, distinctUntilChanged } from "rxjs/operators";
 import { monitorBackboneModel } from "utils/observable/monitorBackboneModel";
 
-// prettier-ignore
-export function syncCurrentHistoryToGalaxy(handler) {
-
-    // wait for the current history panel to appear
-    const currentHistoryPanel$ = monitorChange(() => {
-        return getGalaxyInstance()?.currHistoryPanel?.model;
-    });
-
-    // then emit the id each time it changes
-    const result$ = currentHistoryPanel$.pipe(
-        switchMap(model => monitorBackboneModel(model, "id").pipe(
-            pluck("id")
-        ))
+export function syncCurrentHistoryToGalaxy(galaxy$, store) {
+    const historyId$ = galaxy$.pipe(
+        pluck("currHistoryPanel", "model"),
+        filter(Boolean),
+        switchMap((model) => monitorBackboneModel(model).pipe(shareReplay(1))),
+        pluck("id"),
```

Comment:
It looks like maybe we're getting a new backbone model with every history switch, so observing the stale backbone model for changes doesn't work (for the history id case).
I am concluding that from how I was able to get it to work with
```
diff --git a/client/src/components/History/model/syncCurrentHistoryToGalaxy.js b/client/src/components/History/model/syncCurrentHistoryToGalaxy.js
index 412461a225..9084e04f47 100644
--- a/client/src/components/History/model/syncCurrentHistoryToGalaxy.js
+++ b/client/src/components/History/model/syncCurrentHistoryToGalaxy.js
@@ -1,13 +1,14 @@
 // Sync Galaxy store to legacy galaxy current history
 
-import { switchMap, pluck, shareReplay, filter, distinctUntilChanged } from "rxjs/operators";
+import { switchMap, pluck, shareReplay, filter, distinctUntilChanged, tap } from "rxjs/operators";
 import { monitorBackboneModel } from "utils/observable/monitorBackboneModel";
 
 export function syncCurrentHistoryToGalaxy(galaxy$, store) {
     const historyId$ = galaxy$.pipe(
         pluck("currHistoryPanel", "model"),
+        tap((model) => console.log('tapped model', model)),
         filter(Boolean),
-        switchMap((model) => monitorBackboneModel(model).pipe(shareReplay(1))),
+        // switchMap((model) => monitorBackboneModel(model).pipe(shareReplay(1))),
         pluck("id"),
         distinctUntilChanged()
     );
diff --git a/client/src/utils/observable/waitForInit.js b/client/src/utils/observable/waitForInit.js
index 702eb965b7..c519a6c893 100644
--- a/client/src/utils/observable/waitForInit.js
+++ b/client/src/utils/observable/waitForInit.js
@@ -10,7 +10,7 @@ import { map, filter, take, takeUntil } from "rxjs/operators";
 // prettier-ignore
 export function waitForInit(selector, cfg = {}) {
     const {
-        spamTime = 100,
+        spamTime = 1000,
         timeout = 10000,
         isValid = (val) => val !== undefined
     } = cfg;
@@ -18,7 +18,7 @@ export function waitForInit(selector, cfg = {}) {
     return interval(spamTime).pipe(
         map(() => selector()),
         filter(isValid),
-        take(1),
-        takeUntil(timer(timeout))
+        // take(1),
+        // takeUntil(timer(timeout))
     );
 }
```
Does that make sense ?

Reply:
The intent was that it would fire whenever Galaxy changes its currentHistory value, yes.

Thanks for your experiment! Let me carefully look to see if that's what I wanted. Let me make sure that change in behavior doesn't mess with other places I want to use waitForInit, and if so, I'll refactor stuff appropriately.

Reply:
Good catch! I'm now monitoring all the backbone events on the currHistoryPanel object itself and just peeling the id off the model at that time. 

```javascript
const historyId$ = galaxy$.pipe(
    pluck("currHistoryPanel"),
    switchMap((panel) => {
        // relationship between currHistoryPanel and its model is murky, but I don't care, I'm
        // just going to grab the id fresh after every possible event
        return fromEvent(panel, "all").pipe(
            map(() => panel.model.id)
        );
    }),
    filter(Boolean),
    distinctUntilChanged()
);
```

As for your changes in the diff above, looks like all you did was slow down the initial check loop? You could do that by just passing in different durations like so:

```javascript
waitForInit(getGalaxyInstance, { spamTime: 1000 })
```

But that's a side issue, I don't believe that was the actual problem you found.

Reply:
I just increased this because I didn't find out how to get events out of the backbone model, so I put `tap` operators all over the place 😆